.github/workflows/reusable_test.yml

```diff
@@ -0,0 +1,98 @@
+### For the pure soul wishes to move it to another place
+# https://github.com/orgs/community/discussions/9050
+
+name: Testing workflow
+'on':
+  workflow_call:
+    inputs:
+      test_name:
+        description: the value of test type from tests/ci/ci_config.py, ends up as $CHECK_NAME ENV
+        required: true
+        type: string
+      runner_type:
+        description: the label of runner to use
+        required: true
+        type: string
+      run_command:
+        description: the command to launch the check. Usually starts with `cd '$REPO_COPY/tests/ci'`
+        required: true
+        type: string
+      batches:
+        description: how many batches for the test will be launched
+        default: 1
+        type: number
+      checkout_depth:
+        description: the value of the git shallow checkout
+        required: false
+        type: number
+        default: 1
+      submodules:
+        description: if the submodules should be checked out
+        required: false
+        type: boolean
+        default: false
+      additional_envs:
+        description: additional ENV variables to setup the job
+        type: string
+
+env:
+  CHECK_NAME: ${{inputs.test_name}}
+
+jobs:
+  PrepareStrategy:
+    if: ${{inputs.batches > 0}}  # batches < 1 is misconfiguration
+    runs-on: [self-hosted, style-checker-aarch64]
+    outputs:
+      batches: ${{steps.batches.outputs.batches}}
+    steps:
+      - name: Calculate batches
+        id: batches
+        run: |
+          batches_output=$(python3 -c 'import json; print(json.dumps(list(range(${{inputs.batches}}))))')
```

Comment:
I believe we can avoid using separate runner for this

Reply:
Nope, can't. You are free to fork https://github.com/Felixoid/actions-experiments/commit/69e917383d4e0bf5f44b4cdb8987e490b3e22651 and play, it's the minimum reproducible thing

Reply:
although, we could try to output it from JS action. It could work out

Reply:
But no, even our [debug](https://github.com/ClickHouse/ClickHouse/blob/master/.github/workflows/debug.yml) launches "somewhere"

Reply:
BTW, I thought it will be used to calculate if it's necessarily to run the following tests :wink: